trino/sqlalchemy/dialect.py

```diff
@@ -0,0 +1,301 @@
+            kwargs['catalog'] = db_parts[0]
+            kwargs['schema'] = db_parts[1]
+        else:
+            raise ValueError(f'Unexpected database format {url.database}')
+
+        username = kwargs.pop('username', 'anonymous')
+        kwargs['user'] = username
+
+        password = kwargs.pop('password', None)
+        if password:
+            kwargs['http_scheme'] = 'https'
+            kwargs['auth'] = BasicAuthentication(username, password)
+
+        return args, kwargs
+
+    def get_columns(self, connection: Connection,
+                    table_name: str, schema: str = None, **kw) -> List[Dict[str, Any]]:
+        if not self.has_table(connection, table_name, schema):
+            raise exc.NoSuchTableError(f'schema={schema}, table={table_name}')
+        return self._get_columns(connection, table_name, schema, **kw)
+
+    def _get_columns(self, connection: Connection,
+                     table_name: str, schema: str = None, **kw) -> List[Dict[str, Any]]:
+        schema = schema or self._get_default_schema_name(connection)
+        query = dedent('''
+            SELECT
+                "column_name",
+                "column_default",
+                "is_nullable",
+                "data_type"
+            FROM "information_schema"."columns"
+            WHERE "table_schema" = :schema AND "table_name" = :table
+            ORDER BY "ordinal_position" ASC
+        ''').strip()
+        res = connection.execute(sql.text(query), schema=schema, table=table_name)
+        columns = []
+        for record in res:
+            column = dict(
+                name=record.column_name,
+                type=datatype.parse_sqltype(record.data_type),
+                nullable=(record.is_nullable or '').upper() == 'YES',
+                default=record.column_default,
+            )
+            columns.append(column)
+        return columns
+
+    def get_pk_constraint(self, connection: Connection,
+                          table_name: str, schema: str = None, **kw) -> Dict[str, Any]:
+        """Trino has no support for primary keys. Returns a dummy"""
+        return dict(name=None, constrained_columns=[])
+
+    def get_primary_keys(self, connection: Connection,
+                         table_name: str, schema: str = None, **kw) -> List[str]:
+        pk = self.get_pk_constraint(connection, table_name, schema)
+        return pk.get('constrained_columns')
+
+    def get_foreign_keys(self, connection: Connection,
+                         table_name: str, schema: str = None, **kw) -> List[Dict[str, Any]]:
+        """Trino has no support for foreign keys. Returns an empty list."""
+        return []
+
+    def get_schema_names(self, connection: Connection, **kw) -> List[str]:
+        query = 'SHOW SCHEMAS'
+        res = connection.execute(sql.text(query))
+        return [row.Schema for row in res]
+
+    def get_table_names(self, connection: Connection, schema: str = None, **kw) -> List[str]:
+        query = 'SHOW TABLES'
```

Comment:
Should we use `information_schema` everywhere instead?

Reply:
Is there any reason to avoid using `SHOW` command?

Reply:
When `schema=None`, `SELECT * FROM information_schema.tables` return tables in all schemas but what we wan't is get tables in current schema.
Furthermore, other dialects like [MySQL](https://github.com/sqlalchemy/sqlalchemy/blob/rel_1_3_24/lib/sqlalchemy/dialects/mysql/base.py#L2734) also use `SHOW` command. I don't see any problem if we are going to the same direction.